agnpy/time_evolution/_time_evolution_utils.py

```diff
@@ -0,0 +1,238 @@
+import numpy as np
+from astropy.units import Quantity
+from pygments.styles import vs
+from astropy.constants import m_e, c, h, e
+from agnpy import InterpolatedDistribution, EmptyDistribution
+from agnpy.time_evolution.types import BinsWithDensities, FnParams, TimeEvaluationResult
+from agnpy.utils.conversion import mec2
+from scipy.interpolate import PchipInterpolator
+
+bin_size_factor = 1e-4
+
+class DistributionToSinglePointCollapseError(Exception):
+    def __init__(self, gamma_point):
+        self.gamma_point = gamma_point
+        super().__init__(f"Unsupported state, cannot create InterpolatedDistribution - distribution collapsed to a single gamma point {gamma_point}")
```

Comment:
for now you can leave it as it is with an exception, but if it turns out that we have such cases when dealing with data we can always make a dummy distribution with two points very close to one another - then the index does not matter and the emission would be the virtually same